recipes/libdwarf/all/conanfile.py

```diff
@@ -47,7 +47,8 @@ def layout(self):
         cmake_layout(self, src_folder="src")
 
     def requirements(self):
-        self.requires("libelf/0.8.13")
+        if self.options.with_dwarfgen or self.version == "20191104":
```

Comment:
That's bad. The options `with_dwarfgen` should be `True` by default:

CMakeLists.txt:203

    option(DWARF_WITH_LIBELF "Use libelf (default is YES)" TRUE)


I would suggest changing the default value for `with_dwarfgen` to `True`. I may break some users, but then we will follow the correct behavior.

Reply:
> The options `with_dwarfgen` should be True by default

I guess I disagree, the option to enable `dwarfgen` is False by default in the CMakeList:

CMakeLists.txt:353
```
option(BUILD_DWARFGEN "Build dwarfgen (default is NO)" FALSE)
```

dwarfgen is more a utility for testing dwarf-generation by libdwarf and as an example of how one uses libdwarf to generate DWARF. I don't think that's something you want to get by default if you only want to use the library.

I made the change based on the release notes from v0.1.0, you can check them here in the Releases Section https://www.prevanders.net/dwarf.html. `libelf` is only needed when building `dwarfgen`.

Also, recent changes upstream are removing the remaining references to `libelf`:
https://github.com/davea42/libdwarf-code/commit/77d9cad797963942497d004103b602d55f5a7882
https://github.com/davea42/libdwarf-code/commit/a5d44c690636aa9c49ba4c6656f042c05f42c7ed

Reply:
In that case, if you pass DWARF_WITH_LIBELF=False, would work for the current error, or do you need another patch?

Reply:
if you mean the build errors in this PR, I can change that second commit to not call `find_package(libelf REQUIRED)` instead of just dropping the `REQUIRED`

Reply:
>  I can change that second commit to not call find_package(libelf REQUIRED) instead of just dropping the REQUIRED

@cesenaLA It sounds perfect for the current situation. Could you please update it?

Reply:
I made the change, but it turns out that was not the issue. libdwarf started exporting the libdwarf.pc file in https://github.com/davea42/libdwarf-code/commit/7368a1062271e08e7df0f188d8b02c9991a0d03a. I removed installing that file to the patch.